From 601e5ea89b1fd6fb8047ea7c23eae6eb1c90145f Mon Sep 17 00:00:00 2001
From: John Crispin <blogic@openwrt.org>
Date: Tue, 21 Feb 2012 14:25:03 +0100
Subject: [PATCH 20/70] MIPS: lantiq: helper functions for SoC detection

Add additional functions for runtime soc detection. We need these for the
serial driver.

Signed-off-by: John Crispin <blogic@openwrt.org>
---
 .../include/asm/mach-lantiq/falcon/lantiq_soc.h    |   16 ++++++++++++++--
 .../mips/include/asm/mach-lantiq/xway/lantiq_soc.h |    5 +++++
 2 files changed, 19 insertions(+), 2 deletions(-)

--- a/arch/mips/include/asm/mach-lantiq/falcon/lantiq_soc.h
+++ b/arch/mips/include/asm/mach-lantiq/falcon/lantiq_soc.h
@@ -134,8 +134,20 @@ extern int ltq_gpio_mux_set(unsigned int
 #define LTQ_EIU_BASE_ADDR	0
 #define LTQ_EBU_PCC_ISTAT	0
 
-#define ltq_is_ar9()	0
-#define ltq_is_vr9()	0
+static inline int ltq_is_ar9(void)
+{
+	return 0;
+}
+
+static inline int ltq_is_vr9(void)
+{
+	return 0;
+}
+
+static inline int ltq_is_falcon(void)
+{
+	return 1;
+}
 
 #endif /* CONFIG_SOC_FALCON */
 #endif /* _LTQ_XWAY_H__ */
--- a/arch/mips/include/asm/mach-lantiq/xway/lantiq_soc.h
+++ b/arch/mips/include/asm/mach-lantiq/xway/lantiq_soc.h
@@ -164,5 +164,10 @@ static inline int ltq_is_vr9(void)
 	return (ltq_get_soc_type() == SOC_TYPE_VR9);
 }
 
+static inline int ltq_is_falcon(void)
+{
+	return 0;
+}
+
 #endif /* CONFIG_SOC_TYPE_XWAY */
 #endif /* _LTQ_XWAY_H__ */
